pkg/apitype/core.go

```diff
@@ -323,6 +323,9 @@ type ConfigValue struct {
 	String string `json:"string"`
 	// Secret is true if this value is a secret and false otherwise.
 	Secret bool `json:"secret"`
+	// Object is true if this value is a JSON encoded object. If both `Object` and `Secret` is true,
+	// then the JSON encoded object contains at least one secure value.
+	Object bool `json:"object"`
```

Comment:
It may actually be best to tease out the change to this file to a separate PR that can be merged first. Then update the service to rely on that change. Then merge this PR.

Reply:
#3343